src/iosrv/tests/drpc_client_tests.c

```diff
@@ -235,6 +240,296 @@ test_drpc_verify_notify_bio_error(void **state)
+	assert_string_equal(req->event->ctl_op, cop);
+	assert_string_equal(req->event->str_info, data);
+
+	/* Cleanup */
+	shared__cluster_event_req__free_unpacked(req, NULL);
+	drpc__call__free_unpacked(call, NULL);
+}
+
+static void
+test_drpc_verify_notify_ras(void **state)
+{
+	uuid_t		puuid, cuuid;
+	d_rank_t	rank = 1;
+	daos_obj_id_t	oid = { .hi = 1, .lo = 1 };
+
+	mock_valid_drpc_resp_in_recvmsg(DRPC__STATUS__SUCCESS);
+	assert_int_equal(drpc_init(), 0);
+
+	assert_int_equal(uuid_parse("11111111-1111-1111-1111-111111111111",
+				    puuid), 0);
+	assert_int_equal(uuid_parse("22222222-2222-2222-2222-222222222222",
+				    cuuid), 0);
+
+	ds_notify_ras_event(RAS_RANK_NO_RESPONSE, "no response", RAS_TYPE_INFO,
+			    RAS_SEV_WARN, "exhwid", &rank, "exjobid", &puuid,
+			    &cuuid, &oid, "exctlop",
+			    "{\"people\":[\"bill\",\"steve\",\"bob\"]}");
+	verify_notify_ras(RAS_RANK_NO_RESPONSE, "no response", RAS_TYPE_INFO,
+			  RAS_SEV_WARN, "exhwid", &rank, "exjobid", &puuid,
+			  &cuuid, &oid, "exctlop",
+			  "{\"people\":[\"bill\",\"steve\",\"bob\"]}");
+
+	drpc_fini();
+}
+
+static void
+verify_notify_ras_min_viable(ras_event_t id, char *msg, ras_type_t type,
+			     ras_sev_t sev)
+{
+	Drpc__Call		*call;
+	Shared__ClusterEventReq	*req;
+
+	call = drpc__call__unpack(NULL, sendmsg_msg_iov_len,
+				  sendmsg_msg_content);
+	assert_non_null(call);
+	assert_int_equal(call->module, DRPC_MODULE_SRV);
+	assert_int_equal(call->method, DRPC_METHOD_SRV_CLUSTER_EVENT);
+
+	/* Verify payload contents */
+	req = shared__cluster_event_req__unpack(NULL, call->body.len,
+						call->body.data);
+	assert_non_null(req);
+
+	assert_string_equal(req->event->hostname, dss_hostname);
+	/* populated by mock dss_self_rank */
+	assert_int_equal(req->event->rank, mock_self_rank);
+	assert_int_equal((uint32_t)req->event->id, (uint32_t)id);
+	assert_string_equal(req->event->msg, msg);
+	assert_int_equal((uint32_t)req->event->type, (uint32_t)type);
+	assert_int_equal((uint32_t)req->event->severity, (uint32_t)sev);
+	assert_string_equal(req->event->hw_id, "");
+	assert_string_equal(req->event->job_id, "");
+	assert_string_equal(req->event->pool_uuid, "");
+	assert_string_equal(req->event->cont_uuid, "");
+	assert_string_equal(req->event->obj_id, "");
+	assert_string_equal(req->event->ctl_op, "");
+	assert_true(req->event->str_info == NULL);
+
+	/* Cleanup */
+	shared__cluster_event_req__free_unpacked(req, NULL);
+	drpc__call__free_unpacked(call, NULL);
+}
+
+static void
+test_drpc_verify_notify_ras_min_viable(void **state)
+{
+	mock_valid_drpc_resp_in_recvmsg(DRPC__STATUS__SUCCESS);
+	assert_int_equal(drpc_init(), 0);
+
+	ds_notify_ras_event(RAS_RANK_DOWN, "rank down", RAS_TYPE_STATE_CHANGE,
+			    RAS_SEV_ERROR, NULL, NULL, NULL, NULL, NULL, NULL,
+			    NULL, NULL);
+	verify_notify_ras_min_viable(RAS_RANK_DOWN, "rank down",
+				     RAS_TYPE_STATE_CHANGE, RAS_SEV_ERROR);
+
+	drpc_fini();
+}
+
+static void
+verify_notify_ras_inval()
```

Comment:
This seems to be an exact duplicate of verify_notify_pool_svc_update_inval. If you give the former a more generic name, could use it in both sets of tests and remove this function.